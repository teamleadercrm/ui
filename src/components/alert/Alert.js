import React from 'react';
import { DialogBase } from '../dialog';
import PropTypes from 'prop-types';
import { Heading3 } from '../typography';
import { Box } from '../box';
import { Button } from '../button';
import omit from 'lodash.omit';

const Alert = ({ children, primaryAction, secondaryAction, title, ...otherProps }) => {
  const restProps = omit(otherProps, [
    'primaryAction',
    'secondaryAction',
  ]);

  return (
    <DialogBase  {...restProps} scrollable={false} size="small">
      <Box alignItems="center" display="flex" flexDirection="column" padding={4}>
        <Heading3 color="teal" marginBottom={2}>{title}</Heading3>
        {children}
        <Button level="primary" {...primaryAction} fullWidth marginTop={5} />
        {secondaryAction && <Button {...secondaryAction} fullWidth marginTop={2} />}
      </Box>
    </DialogBase>
  );
};

Alert.propTypes = {
  /** If true, the alert will show on screen. */
  active: PropTypes.bool,
  /** The content to display inside the alert. */
  children: PropTypes.any,
  /** Object containing the props of the primary action (a Button, with level prop set to 'primary'). */
  primaryAction: PropTypes.object.isRequired,
  /** Object containing the the props of the secondary action (a Button). */
  secondaryAction: PropTypes.object,
  /** The title of the alert. */
  title: PropTypes.string,
};

export default Alert;
