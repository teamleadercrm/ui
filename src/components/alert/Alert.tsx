import React from 'react';
import { DialogBase } from '../dialog';
import { Heading3, TextBody } from '../typography';
import { Box } from '../box';
import { Button, BUTTON_LEVELS } from '../button';
import { Illustration60X60Error, Illustration60X60Info } from '@teamleader/ui-illustrations';
import { DialogBaseProps } from '../dialog/DialogBase';
import { ButtonProps } from '../button/Button';

type Action = Omit<ButtonProps, 'fullWidth' | 'marginTop'>;

type Type = 'confirm' | 'destructive' | 'error';

interface AlertProps extends Omit<DialogBaseProps, 'scrollable' | 'size'> {
  /** Object containing the props of the primary action (a Button, with level prop set to 'primary'). */
  primaryAction: Action;
  /** Object containing the the props of the secondary action (a Button). */
  secondaryAction?: Action;
  /** The title of the alert. */
  title: string;
  /** The body of the alert. */
  body?: string;
  /** The type of the alert. */
  type?: Type;
}

const Alert = ({ primaryAction, secondaryAction, title, body, type = 'confirm', ...otherProps }: AlertProps) => {
  return (
    <DialogBase {...otherProps} scrollable={false} size="small">
      <Box alignItems="center" display="flex" flexDirection="column" padding={4} textAlign="center">
        <Box marginBottom={3} marginTop={2}>
          {type === 'confirm' ? <Illustration60X60Info /> : <Illustration60X60Error />}
        </Box>
        <Heading3 color="teal">{title}</Heading3>
        {body && (
          <TextBody color="teal" marginTop={2}>
            {body}
          </TextBody>
        )}
        <Button
          level={type === 'destructive' ? BUTTON_LEVELS.destructive : BUTTON_LEVELS.primary}
          {...primaryAction}
          fullWidth
          marginTop={5}
        />
        {secondaryAction && <Button {...secondaryAction} fullWidth marginTop={2} />}
      </Box>
    </DialogBase>
  );
};

export default Alert;
