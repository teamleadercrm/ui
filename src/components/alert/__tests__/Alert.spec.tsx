import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../Alert';

describe('Component - Alert', () => {
  it('renders', async () => {
    const screen = render(
      <Alert
        active
        primaryAction={{
          label: 'Confirm',
        }}
        secondaryAction={{
          label: 'Cancel',
        }}
        title="Alert title"
        body="I am the alert body text"
      />,
    );

    expect(screen.baseElement).toMatchSnapshot();
  });
});
