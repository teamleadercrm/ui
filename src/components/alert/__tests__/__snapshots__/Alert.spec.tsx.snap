// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Alert renders 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    class="overlay dark overlay"
    data-teamleader-ui="overlay"
  >
    <div>
      <div>
        <div
          tabindex="0"
        />
        <div
          class="box dialog-base is-small is-entering"
          data-teamleader-ui="dialog"
        >
          <div
            class="inner"
          >
            <div
              class="box align-items-center display-flex flex-direction-column padding-bottom-4 padding-left-4 padding-right-4 padding-top-4 text-align-center"
            >
              <div
                class="box margin-bottom-3 margin-top-2"
              >
                <svg
                  fill="currentColor"
                  height="60"
                  preserveAspectRatio="xMidYMid meet"
                  style="vertical-align: middle;"
                  viewBox="undefined"
                  width="60"
                >
                  <path
                    d="M50 30c0 11.046-8.954 20-20 20s-20-8.954-20-20 8.954-20 20-20 20 8.954 20 20z"
                    fill="#0071F2"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M30 26a2 2 0 012 2v11a2 2 0 11-4 0V28a2 2 0 012-2zm0-7a2 2 0 110 4 2 2 0 010-4z"
                    fill="#fff"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
              <h3
                class="box heading heading-3 darkest teal"
                data-teamleader-ui="heading"
              >
                Alert title
              </h3>
              <p
                class="box margin-top-2 text text-body darkest teal"
                data-teamleader-ui="text"
              >
                I am the alert body text
              </p>
              <button
                class="box margin-top-5 reset-box-sizing reset-font-smoothing button-base button primary medium is-full-width"
                data-teamleader-ui="button"
                type="button"
              >
                <span
                  class="box ui-text ui-text-body darkest overflow-singleline"
                  data-teamleader-ui="ui-text"
                >
                  Confirm
                </span>
              </button>
              <button
                class="box margin-top-2 reset-box-sizing reset-font-smoothing button-base button secondary medium is-full-width"
                data-teamleader-ui="button"
                type="button"
              >
                <span
                  class="box ui-text ui-text-body darkest overflow-singleline"
                  data-teamleader-ui="ui-text"
                >
                  Cancel
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          tabindex="0"
        />
      </div>
    </div>
  </div>
</body>
`;
